.github/workflows/build.yml

```diff
@@ -27,6 +27,13 @@ jobs:
       with:
         path: am
 
+    - name: macOS Unshallow
+      if: runner.os == 'macOS'
+      run: |
+        cd am
+        git fetch --unshallow
+        cd ..
```

Comment:
There is already a MacOS specific step (brew etc ...) + try `git -C am --unshallow` instead of changing directories

Reply:
Moved to "Prepare macOS artifacts" and removed "cd"